lib/platform/github/index.ts

```diff
@@ -215,12 +216,28 @@ export async function initRepo({
   config.isGhe = URL.parse(defaults.endpoint).host !== 'api.github.com';
   config.renovateUsername = renovateUsername;
   [config.repositoryOwner, config.repositoryName] = repository.split('/');
-  let res;
+  let repo: GhRepo;
   try {
-    res = await githubApi.getJson<{ fork: boolean }>(`repos/${repository}`);
-    logger.trace({ repositoryDetails: res.body }, 'Repository details');
+    repo = await githubApi.queryRepo<GhRepo>(
+      `{
+      repository(owner: "${config.repositoryOwner}", name: "${config.repositoryName}") {
+        isFork
+        isArchived
+        nameWithOwner
+        mergeCommitAllowed
+        rebaseMergeAllowed
+        squashMergeAllowed
+        defaultBranchRef {
+          name
+        }
+      }
+    }`,
+      {
+        acceptHeader: 'application/vnd.github.v3+json',
```

Comment:
I don't like having this here as it shouldn't be necessary

Reply:
Also need to make 100% sure that the app header gets sent instead when in app mode